packages/tools/grub/package.mk

```diff
@@ -8,11 +8,27 @@ PKG_ARCH="x86_64"
 PKG_LICENSE="GPLv3"
 PKG_SITE="https://www.gnu.org/software/grub/index.html"
 PKG_URL="http://git.savannah.gnu.org/cgit/grub.git/snapshot/${PKG_NAME}-${PKG_VERSION}.tar.gz"
-PKG_DEPENDS_TARGET="toolchain flex freetype:host gettext:host"
+PKG_DEPENDS_HOST="toolchain:host"
+PKG_DEPENDS_TARGET="toolchain flex freetype:host gettext:host grub:host"
 PKG_DEPENDS_UNPACK="gnulib"
 PKG_LONGDESC="GRUB is a Multiboot boot loader."
 PKG_TOOLCHAIN="configure"
 
+pre_configure_host() {
+  unset CFLAGS
+  unset CPPFLAGS
+  unset CXXFLAGS
+  unset LDFLAGS
+  unset CPP
+
+  cd ${PKG_BUILD}
+    # keep grub synced with gnulib
+    ./bootstrap --gnulib-srcdir=$(get_build_dir gnulib) --copy --no-git --no-bootstrap-sync --skip-po
+
+  mkdir -p .${HOST_NAME}
```

Comment:
if you use `pre_build_host()` rather than `pre_configure_host()` you shouldn't need to make the directory and cd into it. The same goes for target.

Reply:
True. But it also needs to set PKG_CONFIGURE_SCRIPT because it is created by pre_build_*() function.

Reply:
> True. But it also needs to set PKG_CONFIGURE_SCRIPT because it is created by pre_build_*() function.

I don't follow.

`PKG_CONFIGURE_SCRIPT` is defined before `pkg_build_*()` is called. Then `PKG_CONFIGURE_SCRIPT` is checked after `pkg_build*()` is called and will be used if `${PKG_BUILD}/configure` is found. No?

Reply:
Ah, seems my package was messed up somehow. Now it works without extra specifying this value. And I was wondering why it needs to be set...

Reply:
> Ah, seems my package was messed up somehow. Now it works without extra specifying this value. And I was wondering why it needs to be set...

It's funny how much extra we do to remove just a few extra lines 😃